src/storage.rs

```diff
@@ -72,4 +83,87 @@ impl BeerTally for HashMapBeerTally {
             },
         }
     }
+
+    fn get_players_vect(&mut self, chat_id: i64) -> Result<Vec<tally::UserTally>, &str> { 
+        if let Some(chat) = self.players.get_mut(&chat_id) {
+            if chat.is_empty() {
+                // Found the chat, but no users are registered
+                // Someone probably unregistered
+                Err("No users have been initialized in this chat, try to register some users first")
+            } else {
+                Ok(chat.values().cloned().collect())
+            }
+        } else {
+            // The chat could not be found (no one has registered yet)
+            Err("No users have been registered yet")
+        }
+    }
+
+    fn get_player(&mut self, chat_id: i64, user_id: i64) -> Result<&mut tally::UserTally, &str> { 
```

Comment:
Maybe a better return type would be `Option<&mut tally::UserTally>`? Then the method which calls this internal method can determine how to handle if the player is not found (e.g. what return string to use).